Makefile

```diff
@@ -10,15 +10,54 @@ CFLAGS_WARNINGS?=-Wall -W -Waggregate-return -Wbad-function-cast -Wcast-align -W
 LIB?=-lm
 SRC=rtl_wmbus.c
 
-all: $(OUTDIR) release
+$(shell $(MKDIR) -p $(OUTDIR))
 
-$(OUTDIR):
-	$(MKDIR) -p "$(OUTDIR)"
+# Create a version number based on the latest git tag.
+COMMIT_HASH?=$(shell git log --pretty=format:'%H' -n 1)
+TAG?=$(shell git describe --tags)
+BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
+CHANGES?=$(shell git status -s | grep -v '?? ')
+TAG_COMMIT_HASH?=$(shell git show-ref --tags | grep $(TAG) | cut -f 1 -d ' ')
 
-release: $(OUTDIR)
+ifeq ($(BRANCH),master)
+  BRANCH:=
+else
+  BRANCH:=$(BRANCH)_
+endif
+
+ifeq ($(COMMIT),$(TAG_COMMIT))
```

Comment:
Neither `COMMIT` nor `TAG_COMMIT` was used previously. Did you mean `COMMIT_HASH` and `TAG_COMMIT_HASH` respectively?

Reply:
So, I copied some code from my wmbusmeters makefile, then I discovered that it had some bugs.
So I removed it. Git itself generates a nice tag string, no need for checking if it is on a tagged commit.